### PR TITLE
[ltm-1.6] Import mode for existing data (Leo)

### PR DESCRIPTION
**Pull request by Kircheneer**
Going off of #178, addressing most of the comments there but making some more drastic changes, like excluding `!create` from usage with import mode.

### CODE REVIEW

File: docs/user/design_lifecycle.md [hunk 1]
```diff
@@ -83,6 +83,32 @@ The decommissioning feature takes into account potential dependencies between de
...
 
 ![design-deployment-decommissioning](../images/screenshots/design-deployment-decommissioning.png)
+
+### Design Deployment Import
+
+Design Builder helps in greenfield use cases (i.e., creating a new data from a design) and in brownfield ones too (i.e., importing existing data that are are related to a new deployment). In the "deployment" mode, a Design Deployment tracks all the objects and attributes that are "owned" by it. With the import functionality, orphan objects and attributes will be incorporated to a new Design Deployment as if they have been set by it.
+
+The import logic works like this:
+
+1. If the object that we reference doesn't exist, normal design creation logic applies.
```
Comment (Kircheneer) [inline docs/user/design_lifecycle.md:98]: This needs to be updated still

File: nautobot_design_builder/design_job.py [hunk 2]
```diff
@@ -77,10 +77,22 @@ def determine_deployment_name(cls, data):
         deployment_name_field = cls.deployment_name_field()
         if deployment_name_field is None:
             if "deployment_name" not in data:
-                raise DesignImplementationError("No instance name was provided for the deployment.")
+                raise DesignImplementationError("No Deployment name was provided for the deployment.")
             return data["deployment_name"]
         return data[deployment_name_field]
 
+    @classmethod
+    def determine_import_mode(cls, data):
```
Comment (Kircheneer) [inline nautobot_design_builder/design_job.py:85]: Haven't addressed this yet from the original comments

File: nautobot_design_builder/models.py [hunk 3]
```diff
@@ -368,13 +373,44 @@ def log(self, model_instance):
...
+            # we acknowledge it and set `full_control` to `False`.
+            # TODO: Shouldn't this change record object also need to be active?
+            change_records_for_instance = ChangeRecord.objects.filter_by_design_object_id(_design_object_id=instance.id)
+            if model_instance.metadata.action == ModelMetadata.CREATE_OR_UPDATE:
+                entry_parameters["full_control"] = not change_records_for_instance.filter(full_control=True).exists()
+
+            # When we don't want to assume full control, make sure we don't try to own any of the query filter values.
+            if not entry_parameters["full_control"]:
+                for attribute in model_instance.metadata.query_filter_values:
+                    entry_parameters["changes"].pop(attribute, None)
```
Comment (abates) [inline nautobot_design_builder/models.py:403]: I don't understand what is happening here, can the code comment be clarified, please?

File: nautobot_design_builder/models.py [hunk 4]
```diff
@@ -368,13 +373,44 @@ def log(self, model_instance):
...
+
+            # Check if any owned attributes exist that conflict with the changes for this instance.
+            for record in change_records_for_instance:
+                for attribute in record.changes:
+                    if attribute in entry_parameters["changes"]:
+                        raise ValueError(  # pylint: disable=raise-missing-from
+                            f"The {attribute} attribute for {instance} is already owned by Design Deployment {record.change_set.deployment}"
+                        )
```
Comment (abates) [inline nautobot_design_builder/models.py:411]: Also here, I'm not sure what is happening... perhaps a concrete example in code comments?

**Review by abates — APPROVED** [review]

**Comment by abates** [comment]
LGTM

**Pull request merged**